### ISSUE TITLE
Issue with Owned DLC Detection and Incorrect Links in CheckDLC Plugin

### ISSUE THREAD

**Issue by Mike-Aniki** [Lacro59/playnite-checkdlc-plugin#83]
Hello,

I’ve been using your CheckDLC plugin to manage and display the DLCs for my games, but I’ve encountered an issue that seems to affect several users. The plugin is supposed to detect the DLCs I own for my games, but unfortunately, all DLCs are marked as "not owned," even for games where I do own DLCs. Based on feedback from other users on Discord, it seems like I’m not the only one experiencing this problem.

Additionally, I’ve noticed that the DLC links generated by the plugin are incorrect. Instead of leading to the relevant DLC pages, the links redirect to api.steampowered.com, which might be contributing to the issue or may indicate a related problem with the API calls.

I’m not sure if these two issues are connected, but I wanted to bring both to your attention. If there's any additional information I can provide to help troubleshoot, feel free to ask.

**Comment by Lacro59** [comment]
Can you try this version?
[playnite-checkdlc-plugin_1_1.zip](https://github.com/user-attachments/files/17329781/playnite-checkdlc-plugin_1_1.zip)

**Comment by Mike-Aniki** [comment]
I have installed and tested it, and the issue with the DLC links has been resolved.
The links now correctly lead to the Steam pages of the relevant DLCs, which is great. However, the problem with detecting owned DLCs still persists, so it wasn’t related to the incorrect link redirection.

If I can suggest a temporary solution and if it's possible, it would be to add an option for users to manually mark DLC as owned. This would be a temporary solution until the detection problem is fully resolved...

Thank you again for your work and support

**Comment by Lacro59** [comment]
Can you post your extenstion.log?

**Comment by Mike-Aniki** [comment]
I've refreshed the dlc for Elden Ring, then I've deleted the dlc for like a dragon: Ishin and I've refreshed the data.

```
11-10 12:40:17.974|DEBUG|Extras#ThemeExtras:Pushed "Affichage en grille - ELDEN RING"
11-10 12:40:20.077|INFO |CheckDlc#CheckDlcDatabase:RefreshNoLoader(ELDEN RING - 3fc72462-85bf-43e9-85d2-7437ba6a667a)
11-10 12:40:20.077|INFO |CheckDlc#SteamDlc:Get dlc for ELDEN RING with Steam
11-10 12:40:22.820|INFO |CheckDlc#SteamDlc:Find 2 dlc
11-10 12:40:57.243|DEBUG|Extras#ThemeExtras:Pushed "Affichage en grille - Like a Dragon: Ishin!"
11-10 12:41:05.580|INFO |CheckDlc#CheckDlcDatabase:RefreshNoLoader(Like a Dragon: Ishin! - 2cc76f9d-4658-4e6a-bd94-c9ca2b7a17ac)
11-10 12:41:05.580|INFO |CheckDlc#SteamDlc:Get dlc for Like a Dragon: Ishin! with Steam
11-10 12:41:08.969|INFO |CheckDlc#SteamDlc:Find 5 dlc

```

For example, here's a screenshot from steam to confirm that I have the elden ring dlc :

![image](https://github.com/user-attachments/assets/4b81a06a-2bf8-4c98-9fc9-1dc7d1ccc835)

And here's a screenshot from CheckDLC :

![image](https://github.com/user-attachments/assets/06486de9-84f7-4d6f-a359-daf439ad6c14)

__________________________________________________________________________________________________

I've noticed that DLC for EPIC Games are well detected but the log is the same as for steam games which don't detect purchased dlc, in the following example I deleted the Saint Row data then refreshed the data :

```
11-10 13:16:59.478|INFO |CheckDlc#CheckDlcDatabase:RefreshNoLoader(Saints Row - be56ef21-48e3-418c-9d3f-f245e6d44852)
11-10 13:16:59.478|INFO |CheckDlc#EpicDlc:Get dlc for Saints Row with Epic
11-10 13:17:07.063|INFO |CheckDlc#EpicDlc:Find 15 dlc
11-10 13:17:22.499|INFO |CheckDlc#CheckDlcDatabase:RefreshNoLoader(Saints Row - be56ef21-48e3-418c-9d3f-f245e6d44852)
11-10 13:17:22.499|INFO |CheckDlc#EpicDlc:Get dlc for Saints Row with Epic

```
![image](https://github.com/user-attachments/assets/01a7cd5c-5736-46f3-8b12-fa3337252a69)

**Comment by Lacro59** [comment]
Thanks for the feedback.
Can you try this version?
[playnite-checkdlc-plugin_1_1.zip](https://github.com/user-attachments/files/17343567/playnite-checkdlc-plugin_1_1.zip)

**Comment by Mike-Aniki** [comment]
It works perfectly with this version ! Thank you for your responsiveness 👍 

Here's my experience:

- I installed this version of the plugin.
- I refreshed the Elden Ring DLCs, but it removed the data as if no DLCs were installed.
- I checked the plugin settings, I saw that I was disconnected from Steam, so I logged back in.
- After that, I refreshed the Elden Ring data, and checkdlc successfully detected all the game's DLCs, including the ones I own!
- I also checked several other games, and everything works perfectly.

I'm attaching a screenshot to show the result.

Just a small note : Icons in the top left don't appear, and there's a "0" on the left of "owned" (see screenshot).

Thanks again for your hard work and quick response to the reported issues !
I also think it would be helpful to mention in the patch notes that users need to reauthenticate with Steam.

![image](https://github.com/user-attachments/assets/01fd720c-28d8-4da9-ab71-196284dfbb99)
